src/pybind/mgr/diskprediction/common/db.py

```diff
@@ -0,0 +1,492 @@
+"""
+Ceph database API
+
+"""
+from __future__ import absolute_import
+
+import json
+import rbd
+import rados
+from mgr_module import CommandResult
+
+
+RBD_FEATURES_NAME_MAPPING = {
+    rbd.RBD_FEATURE_LAYERING: "layering",
+    rbd.RBD_FEATURE_STRIPINGV2: "striping",
+    rbd.RBD_FEATURE_EXCLUSIVE_LOCK: "exclusive-lock",
+    rbd.RBD_FEATURE_OBJECT_MAP: "object-map",
+    rbd.RBD_FEATURE_FAST_DIFF: "fast-diff",
+    rbd.RBD_FEATURE_DEEP_FLATTEN: "deep-flatten",
+    rbd.RBD_FEATURE_JOURNALING: "journaling",
+    rbd.RBD_FEATURE_DATA_POOL: "data-pool",
+    rbd.RBD_FEATURE_OPERATIONS: "operations",
+}
+
+
+def differentiate(data1, data2):
+    """
+    >>> times = [0, 2]
+    >>> values = [100, 101]
+    >>> differentiate(*zip(times, values))
+    0.5
+    """
+    return (data2[1] - data1[1]) / float(data2[0] - data1[0])
+
+
+class DB_API(object):

```

Comment:
This is a confusing name for something that isn't a database.  Perhaps call it "ClusterData" or something like that, to reflect that this is a set of calls that give you the info about a Ceph cluster (and nothing to do with any database)?